### ISSUE TITLE
"<t> is undefined" message appears and the pressed key is ignored

### ISSUE THREAD

**Issue by bjonnh** [syl20bnr/spacemacs#7855]
#### Description :octocat:
"<t> is undefined" message appears and the pressed key is ignored

Any time I switch workspace, and come back to my emacs workspace (using i3), after pressing a key I get the "<t> is undefined" message and the key is ignored. I can type after.

This also happens randomly while writing (in any mode) and also ignore the key that was pressed before the message appears. 

#### Reproduction guide :beetle:
- Start Emacs
- Switch to another workspace
- Type a key
- Switch back to the emacs workspace
- Type a key

*Observed behaviour:* :eyes: :broken_heart:
The message "<t> is undefined" appears and the first key pressed is ignored

*Expected behaviour:* :heart: :smile:
The key should work.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.1.1
- Spacemacs: 0.200.5
- Spacemacs branch: master (rev. 664ba6a)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(python helm auto-completion emacs-lisp git markdown org spell-checking syntax-checking version-control rust)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11

#### Backtrace :paw_prints:

I tried to backtrace on message, that led me to undefined() which didn't led anything more.

**Comment by bjonnh** [comment]
I can reproduce it with layers
```elisp
(helm emacs-lisp)
``` 
only

**Comment by bjonnh** [comment]
Using a X11 key logger, I see that there is no spurious keycode generated, and that they key I pressed is the only thing sent to emacs.

**Comment by d12frosted** [comment]
I experience this problem as well. Can't reproduce it in `emacs -Q`.

**Comment by bjonnh** [comment]
@d12frosted what is your config ? Linux/X11 ? If so what is your WM?

When I'm full screen it doesn't happen in I3 (a tiling window-manager)
so now I'm thinking it is related to the modeline being too small

**Comment by bjonnh** [comment]
So I did C-h m and disabled all the modes by half. And pinned down server-mode as one of the probable culprits…

**Comment by d12frosted** [comment]
I disabled some layers and packages and no longer experience this problem. But still not sure what package / layer / config caused this issue. Once I find it - I'll let you know.

**Comment by Bersam** [comment]
Any news on this? I'm facing same problem. I have i3wm. here's my config file:
https://github.com/Bersam/dotfiles/blob/master/emacs/spacemacs

**Comment by bjonnh** [comment]
Yes, I found a "way" to get rid of it. But it implies disabling the server-mode (which is really annoying)

Just run server-mode twice and it will disable it…

I didn't found anything to solve that, and noticed that it can happen in standard emacs too.

**Comment by Bersam** [comment]
@bjonnh it's only works for once, and after swticthing to another workspace it became same thing again! It's really annoying!

**Comment by bjonnh** [comment]
Interesting, for me it stops it completely. I think we would have to find someone that can build emacs with debug mode and dig where that is coming from. I couldn't find anything in the lisp code.

**Comment by PythonNut** [comment]
[Potentially relevant line in `subr.el`](https://github.com/emacs-mirror/emacs/blob/fb997d30af28da4712ca64876feddbd07db20e13/lisp/subr.el#L765).

**Comment by Bersam** [comment]
@bjonnh I found out that after applying your workaround (server-mode) it occur again when i switch to magit-status, then i remembered that i activated fullscreen mode for magit-status in my .spacemacs 
```elisp
     (git :variables
          git-gutter-use-fringe t
          git-magit-status-fullscreen t)
```
with making git-magit-status-fullscreen to nil, it doens't appear again. I'm guessing that this whole problem is with fullscreen mode in emacs, but making these nil didn't solve the problem:
```elisp
   dotspacemacs-fullscreen-at-startup nil
   dotspacemacs-fullscreen-use-non-native nil
   dotspacemacs-maximized-at-startup nil
```
I thought maybe you got an idea why these are related.

**Comment by omidmnz** [comment]
Same problem here. Any progress?

**Comment by hmc-catfish-s17** [comment]
This appears to be linked to the usage of `sit-for` and [this Emacs bug](https://lists.gnu.org/archive/html/bug-gnu-emacs/2014-05/msg01308.html).

**Comment by PythonNut** [comment]
This appears to be linked to the usage of `sit-for` and [this Emacs bug](https://lists.gnu.org/archive/html/bug-gnu-emacs/2014-05/msg01308.html).

**Comment by mskorzhinskiy** [comment]
I'm not a spacemacs user anymore, but I did encounter the same problem in my setup. I have three different emacs serves and the problem appears only on a server with "org" name.
I've managed to fix this for me changing the server name (server-name variable) from "org" to "ork". I understand, that it's kind of magic, but it helps me, may be it will help someone.

**Comment by mayanez** [comment]
Any follow up on this?

**Comment by ejmg** [comment]
I am actually in the process of moving away from spacemacs and back to vanilla emacs but am experiencing this same behavior as well with my own init file, interestingly enough. In my case, it appears either anaconda-mode or company-mode is causing it. From what I have tested, only python-mode seems to trigger this, leading me to suspect it has something to do with anaconda(?)

**Comment by minimalprocedure** [comment]
I've this with javascript but disabling tern-mode seem to go away. If I type, for example,  'import' text is 'imo..' often with some chars lost. It seems related to the speed of communication with ternjs (or other externals) and text completion system. With the usual configuration the problem appeared about three or four days ago.

**Comment by mayanez** [comment]
It seems to be related to auto-completion somehow. All of my other +lang layers that have it enabled suffer from it. The issue stops once I disable completion in a layer.

**Comment by marchdf** [comment]
Also experiencing this issue... It seems to have popped up pretty recently for me (using my own init file, not spacemacs). If I type slowly enough, I don't see the issue. Seeing this in python-mode (not using anaconda-mode) and the issue disappears when I disable elpy-mode. It definitely feels like a completion communication problem...

**Comment by ejmg** [comment]
I do not have the issue within spacemacs itself but that is because I have
**not updated it** in about 3-4 months.

My init file for my vanilla emacs config is up to date and is the only one
giving me this error.

I have tested company modes related to haskell, lisp, rust, and python with
only python giving me this error.

Is anyone familiar with how we could go about debugging
completion/company-mode? Given we have people reporting JS and Python cases
of this issue, it probably is an error/breaking change upstream from
company itself rather than anaconda or tern mode(?)

**Comment by marchdf** [comment]
Apparently this was fixed by this commit: https://github.com/company-mode/company-mode/commit/259cf40baa24bd0dcb9df743248154b35f81a74e

**Comment by ejmg** [comment]
This did it for me!

**Comment by minimalprocedure** [comment]
the problem seem to go away... :)

**Comment by minimalprocedure** [comment]
ok, no... the problem persist. It's not as pervasive as before but is here.

**Comment by minimalprocedure** [comment]
ok, without spell-checking layer it seem going away.

**Comment by offbyone** [comment]
I'm seeing this in Python mode. Disabling flyspell-mode seems to be the key

**Comment by ngoonee** [comment]
Update your packages, then use list-packages to check whether company is updated to the latest available (currently 20180317.1008 for me). That seems to fix this annoying issue for me.
